projects/alpaca-finance/lyf.js

```diff
@@ -74,25 +80,48 @@ async function calLyfTvl(chain, block) {
     );
 
     /// @dev update balances directly for single-asset workers
-    const singleAssetWorkersBalances = stakingTokenInfos
-      .filter((n) => {
-        /// @dev filter only single-asset LYF workers
-        const name = addresses["Vaults"][i]["workers"].find(
-          (w) => w.address === n.input.params[1]
-        ).name;
-        if (name.includes("CakeMaxiWorker")) {
-          return true;
-        }
-        return false;
+    const singleAssetWorkersInfos = (
+      await sdk.api.abi.multiCall({
+        block,
+        abi: abi.userInfoCake,
+        calls: addresses["Vaults"][i]["workers"].filter((n) => {
+          /// @dev filter only single-asset LYF workers
+          return n.name.includes("CakeMaxiWorker");
+        }).map((worker) => {
+          return {
+            target: worker["stakingTokenAt"],
+            params: [worker["address"]],
+          };
+        }),
+        chain,
       })
+    ).output;
+
+    const singleAssetPrice = (
+      await sdk.api.abi.multiCall({
+        block,
+        abi: abi.singleAssetPrice,
+        calls: addresses["Vaults"][0]["workers"].filter((n) => {
```

Comment:
does this adapter run for you locally? because when I try, I get error. But I am not seeing any bug in your code. 

This adapter works fine before your change

Reply:
no, it doesn't run locally. it will read file JSON. I think that code is an error because file JSON is updated, but the file is already updated.